landlab/components/depression_finder/lake_mapper.py

```diff
@@ -313,24 +291,61 @@ def updated_boundary_conditions(self):
         if self._D8:
             diag_nbrs = self._grid.diagonal_adjacent_nodes_at_node.copy()
             # remove the inactive nodes:
-            diag_nbrs[self.grid.status_at_node[diag_nbrs] == CLOSED_BOUNDARY] = -1
+            diag_nbrs[self._grid.status_at_node[diag_nbrs] == CLOSED_BOUNDARY] = -1
             self._node_nbrs = np.concatenate((self._node_nbrs, diag_nbrs), 1)
             self._link_lengths = np.empty(8, dtype=float)
             self._link_lengths[0] = dx
             self._link_lengths[2] = dx
             self._link_lengths[1] = dy
             self._link_lengths[3] = dy
             self._link_lengths[4:].fill(np.sqrt(dx * dx + dy * dy))
-        elif (type(self.grid) is landlab.grid.raster.RasterModelGrid) and (
-            self._routing == "D4"
-        ):
+        elif isinstance(self._grid, RasterModelGrid) and (self._routing == "D4"):
             self._link_lengths = np.empty(4, dtype=float)
             self._link_lengths[0] = dx
             self._link_lengths[2] = dx
             self._link_lengths[1] = dy
             self._link_lengths[3] = dy
         else:
-            self._link_lengths = self.grid.length_of_link
+            self._link_lengths = self._grid.length_of_link
+
+    @property
+    def is_pit(self):
+        """At node array indicating whether the node is a pit or not."""
+        return self._is_pit
+
+    @property
+    def number_of_pits(self):
+        """The number of pits on the grid."""
+        return self._number_of_pits
+
+    @property
+    def pit_node_ids(self):
+        """Node IDs of grid nodes identified as pits."""
+        return self._pit_node_ids
+
+    @property
+    def flood_status(self):
+        """Map of flood status (_PIT, _CURRENT_LAKE, _UNFLOODED, or _FLOODED)."""
+        return self._flood_status
+
+    @property
+    def receivers(self):
+        """At node array indicating which node receives flow."""
+        return self._receivers
+
+    @receivers.setter
+    def receivers(self, receivers):
+        self._receivers = receivers
+
+    @property
+    def depression_depth(self):
+        """At node array of depression depths."""
+        return self._depression_depth
+
+    @property
+    def depression_outlet_map(self):
+        """At node array indicating the node-id of the depression outlet."""
+        return self._depression_outlet_map
```

Comment:
I'll provide a small bit of background to this and similar changes. 

Previously this component had a public, undocumented attribute `self._depression_outlet_map` and it was featured in a docstring.

Thus in treating this (and similar) instances, I had two choices:

1) Make the attribute private and change the docstring OR
2) Make the attribute private, interpret the use in a docstring as intention by the component developer that such an attribute should exist as a public attribute and thus create a @property for it. 

I went with 2 because it seemed to honor the original developer's intention AND not change the user-facing example. 

I'm not certain how pythonic it is to have two ways to access the same array (through the grid or through the component), but I can say that many components provide access to key fields (and highlight this access in the docstrings) in this way.

Reply:
Ok, that makes sense. I understand that the more general pattern this implies is "leave it to the component author to decide whether or not to offer public attribute variables that are also fields," which seems reasonable to me.

Reply:
@gregtucker @kbarnhart I agree, I think it's fine to leave it up to the component authors. However, I'm not sure we should encourage it. I like the idea of having just a single way to get values from the grid. I could definitely be persuaded though.

🤔 Actually, maybe something along these lines would be a good thing. Not necessarily having fields be properties of components, but properties of grids (e.g. `grid.at_node.depression_outlet_map`. This might reduce the chance of some pesky, hard-to-find errors where one component changes a reference to a field but doesn't tell other components.

Reply:
There are a couple of reason's I would advocate making no changes but advocating to component authors that they access the grid fields through the grid (a recommendation in our Rules and Recommendations)

My reasons:

1) These component attributes have been user facing and we highlight them in our docstrings. Thus making changes may create more problems for users than we'd like. 

2) I think @mcflugen 's idea is nice, with but there is a major barrier to it being functional. The component attribute names and the field names are not always the same. For example, `DepressionFinder.depression_outlet_map` points to `grid.at_node["depression__outlet_node"]`.

Reply:
@kbarnhart Sorry, I was changing the topic. I meant something slightly different. All I meant was using
```python
>>> grid.at_node.depression__outlet_node
```
instead of
```python
>>> grid.at_node["depression__outlet_node"]
```
*depression__outlet_node* would be a property of *grid.at_node*, which would mean a user couldn't easily change it's reference. Components themselves could still create properties with whatever names they like.